dbt/include/snowflake/macros/materializations/dynamic_table/ddl.sql

```diff
@@ -1,46 +1,96 @@
-{% macro snowflake__get_alter_dynamic_table_as_sql(
-    relation,
+{% macro snowflake__alter_dynamic_table_sql(
+    dynamic_table,
     configuration_changes,
-    sql,
     existing_relation,
     backup_relation,
     intermediate_relation
 ) -%}
-    {{- log('Applying ALTER to: ' ~ relation) -}}
-    {{- snowflake__get_replace_dynamic_table_as_sql(relation, sql, existing_relation, backup_relation, intermediate_relation) -}}
+    {{- log('Applying ALTER to: ' ~ dynamic_table.path) -}}
+
+    {% if configuration_changes.requires_full_refresh %}
+
+        {{ snowflake__replace_dynamic_table_sql(dynamic_table, existing_relation, backup_relation, intermediate_relation) }}
+
+    -- otherwise apply individual changes as needed
```

Comment:
```suggestion
    -- otherwise apply the changes collected in configuration_changes
```
or whatever would be most descriptive.

Also, there is an argument a good in-code comment says why rather than what so maybe just a rewrite to
```
    -- a full refresh does not occur _<blah blah>___ 
```

Reply:
> Also, there is an argument a good in-code comment says why rather than what
Agreed. I'll update this.